src/libvirtApi/domain.js

```diff
@@ -99,15 +101,13 @@ function buildConsoleVVFile(consoleDetail) {
         'fullscreen=0\n';
 }
 
-function domainAttachDevice({ connectionName, vmId, permanent, hotplug, xmlDesc }) {
-    let flags = Enum.VIR_DOMAIN_AFFECT_CURRENT;
-    if (hotplug)
-        flags |= Enum.VIR_DOMAIN_AFFECT_LIVE;
-    if (permanent)
-        flags |= Enum.VIR_DOMAIN_AFFECT_CONFIG;
+function domainAttachDevice({ connectionName, vmId, xmlDesc, live }) {
+    let attachflag = Enum.VIR_DOMAIN_AFFECT_CONFIG;
+    if (live) {
+        attachflag |= Enum.VIR_DOMAIN_AFFECT_LIVE;
+    }
 
-    // Error handling is done from the calling side
-    return call(connectionName, vmId, 'org.libvirt.Domain', 'AttachDevice', [xmlDesc, flags], { timeout, type: 'su' });
+    return call(connectionName, vmId, 'org.libvirt.Domain', 'AttachDevice', [xmlDesc, attachflag], { timeout, type: 'su' });
```

Comment:
A lot of test failures https://logs.cockpit-project.org/logs/pull-472-20220114-124851-4e91fab0-debian-testing/log.html. My gut tells me it's probably caused by this change, but will have to investigate futher

Reply:
That is because I modified the this function at the first and missed parameter "permanent", as we don't use this method when attaching device, I revert this method